src/test/logging_tests.cpp

```diff
@@ -15,19 +15,24 @@ BOOST_FIXTURE_TEST_SUITE(logging_tests, BasicTestingSetup)
 BOOST_AUTO_TEST_CASE(logging_timer)
 {
     SetMockTime(1);
-    auto sec_timer = BCLog::Timer<std::chrono::seconds>("tests", "end_msg");
+    auto micro_timer = BCLog::Timer<std::chrono::microseconds>("tests", "end_msg");
     SetMockTime(2);
-    BOOST_CHECK_EQUAL(sec_timer.LogMsg("test secs"), "tests: test secs (1.00s)");
+    BOOST_CHECK_EQUAL(micro_timer.LogMsg("test micros"), "tests: test micros (1000000μs)");
 
     SetMockTime(1);
     auto ms_timer = BCLog::Timer<std::chrono::milliseconds>("tests", "end_msg");
     SetMockTime(2);
     BOOST_CHECK_EQUAL(ms_timer.LogMsg("test ms"), "tests: test ms (1000.00ms)");
 
     SetMockTime(1);
-    auto micro_timer = BCLog::Timer<std::chrono::microseconds>("tests", "end_msg");
+    auto sec_timer = BCLog::Timer<std::chrono::seconds>("tests", "end_msg");
     SetMockTime(2);
-    BOOST_CHECK_EQUAL(micro_timer.LogMsg("test micros"), "tests: test micros (1000000μs)");
+    BOOST_CHECK_EQUAL(sec_timer.LogMsg("test secs"), "tests: test secs (1.00s)");
+
+    SetMockTime(1);
+    auto minute_timer = BCLog::Timer<std::chrono::minutes>("tests", "end_msg");
+    SetMockTime(2);
+    BOOST_CHECK_EQUAL(minute_timer.LogMsg("test minutes"), "Error: unexpected time type");
```

Comment:
seems a bit odd to add a new feature in one commit and then remove it in the next commit.

If you squash the two commits and keep the ordering, the diff will be a lot smaller, if not empty.

Reply:
Kept the last three commits separate as they do three different things and were contributed by three different people.

Reply:
Fair enough

Reply:
3 different commits by 3 different people is irrelevant if we're just adding code to immediately delete it. As long as bddae7e7ff7bb5931ed807acaef7336f2ee98476 came first it didn't matter. So I don't understand why the remaining two commits, 498b323425d960274c40472a6a847afc1982201d and f530202353a4f8bb444966559aa15681ab3cebc6, couldn't have been squashed together and just have the two authors.